kotlin.kt

```diff
@@ -1,20 +1,16 @@
-public fun main(args: Array<String>) {
-	for (i in 1..100) {
-		fizzBuzz(i)
-	}
+inline fun Int.divisibleBy(x: Int) = this % x == 0
+
+fun main(args: Array<String>) {
+	(1..100).forEach(::fizzBuzz)
 }
 
 fun fizzBuzz(n: Int) {
-	val divisibleBy3 = n % 3 == 0
-	val divisibleBy5 = n % 5 == 0
-
-	if (divisibleBy3 && divisibleBy5) {
-		println("FizzBuzz")
-	} else if (divisibleBy5) {
-		println("Buzz")
-	} else if (divisibleBy3) {
-		println("Fizz")
-	} else {
-		println(n)
-	}
+	println(
+		when (true) {
+			n.divisibleBy(15) -> "FizzBuzz"
```

Comment:
There are a lot of ways to use when in Kotlin, here's one more example just to give you an idea

```kotlin
val x = Random().nextInt(3)
when (x) {
  0 -> println("x is 0")
  1 -> println("x is 1")
  2 -> println("x is 2")
  else -> println("x is $x")
}
```

Note that the `else` is required, unless you can prove to the compiler that it's impossible to not fit one of the criteria, like your cases being `n = 0`, `n > 0` and `n < 0`